aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -239,7 +239,7 @@ void parallel_cat(const Tensor &out, const MaterializedITensorListRef& inputs, i
                   int nDims, c10::MemoryFormat memory_format) {
   // First, let's set up our kernel parameters. We start with a raw pointer to
   // the storage for the output Tensor.
-  scalar_t *data = out.mutable_data_ptr<scalar_t>();
+  scalar_t *data = (scalar_t *) out.mutable_data_ptr();
```

Comment:
Nit (also, why are you making change in the first place? Is that because one can not cast to bit types using the template?)
```suggestion
  scalar_t *data = static_cast<scalar_t *>(out.mutable_data_ptr());
```